addons/backgrounds/src/containers/BackgroundSelector.tsx

```diff
@@ -43,85 +65,114 @@ const createBackgroundSelectorItem = memoize(1000)(
+    change: (arg: { selected: string; name: string }) => void
+  ) => {
+    const backgroundSelectorItems = backgrounds.map(({ name, value }) =>
+      createBackgroundSelectorItem(null, name, value, true, change)
+    );
+
+    if (selectedBackgroundColor !== 'transparent') {
+      return [
+        createBackgroundSelectorItem('reset', 'Clear background', 'transparent', null, change),
+        ...backgroundSelectorItems,
+      ];
+    }
+
+    return backgroundSelectorItems;
   }
+);
 
+const getSelectedBackgroundColor = (
+  backgrounds: Background[] = [],
+  currentSelectedValue: string,
+  defaultName: string
+): string => {
   if (currentSelectedValue === 'transparent') {
-    return currentSelectedValue;
+    return 'transparent';
   }
 
-  if (list.find((i) => i.value === currentSelectedValue)) {
+  if (backgrounds.find((background) => background.value === currentSelectedValue)) {
     return currentSelectedValue;
   }
 
-  if (list.find((i) => i.default)) {
-    return list.find((i) => i.default).value;
+  const defaultBackground = backgrounds.find((background) => background.name === defaultName);
+  if (defaultBackground) {
+    return defaultBackground.value;
   }
 
-  return 'transparent';
-};
-
-const mapper = ({ api, state }: Combo): { items: Input[]; selected: string | null } => {
-  const list = api.getCurrentParameter<Input[]>(PARAM_KEY);
-  const selected = state.addons[PARAM_KEY] || null;
+  if (defaultName) {
+    const availableColors = backgrounds.map((background) => background.name).join(', ');
+    logger.warn(
+      `Backgrounds Addon: could not find the default color "${defaultName}".
+      These are the available colors for your story based on your configuration: ${availableColors}`
+    );
+  }
 
-  return { items: list || [], selected };
+  return 'transparent';
 };
 
-const getDisplayedItems = memoize(10)(
-  (
-    list: Input[],
-    selected: string | null,
-    change: (arg: { selected: string; name: string }) => void
-  ) => {
-    let availableBackgroundSelectorItems: Item[] = [];
+const getBackgroundsConfig = ({ api, state }: Combo): BackgroundsConfig => {
+  const backgroundsParameter = api.getCurrentParameter<BackgroundsParameter>(BACKGROUNDS_PARAM_KEY);
+  const selectedBackgroundValue = state.addons[BACKGROUNDS_PARAM_KEY] || null;
 
-    if (selected !== 'transparent') {
-      availableBackgroundSelectorItems.push(
-        createBackgroundSelectorItem('reset', 'Clear background', 'transparent', null, change)
-      );
-    }
-
-    if (list.length) {
-      availableBackgroundSelectorItems = [
-        ...availableBackgroundSelectorItems,
-        ...list.map(({ name, value }) =>
-          createBackgroundSelectorItem(null, name, value, true, change)
-        ),
-      ];
-    }
-
-    return availableBackgroundSelectorItems;
+  if (Array.isArray(backgroundsParameter)) {
+    logger.warn(
+      'Addon Backgrounds api has changed in Storybook 6.0. Please refer to the migration guide.'
```

Comment:
Is this good enough for notifying people to migrate?

Reply:
A link would be nice!

Reply:
Link added. Should point to /next instead of /master, right? In case we have early adopters.